test_invite_nonexistent_user (test_channel_management.TestChannelManagement.test_invite_nonexistent_user)
Tests inviting a non-existent user. ... DEBUG: Client proposed 'InviteSender', server assigned 'anonthe_curiousa' with user 'user_anonthe_curiousa' and host 'localhost'. Full response: ['USER user_anonthe_curiousa 0 * :real_anonthe_curiousa', ':localhost 001 anonthe_curiousa :Welcome to the IRC server anonthe_curiousa!', ':localhost 002 anonthe_curiousa :Your host is localhost, running version 1.0', ':localhost 003 anonthe_curiousa :This server was created today', ':localhost 004 anonthe_curiousa localhost 1.0 o o', ':localhost 375 anonthe_curiousa :- ~ Meowdy, traveler! ~', ":localhost 372 anonthe_curiousa :- Here's your good-luck cat butt:", ':localhost 372 anonthe_curiousa :-     /\\_/\\', ':localhost 372 anonthe_curiousa :-    ( o.o )', ':localhost 372 anonthe_curiousa :-     > ^ <    🍑', ':localhost 372 anonthe_curiousa :-     Cat butt initialized.', ':localhost 376 anonthe_curiousa :End of /MOTD']
Test: anonthe_curiousa inviting GhostUser (should fail)...
ERROR
test_invite_not_on_channel (test_channel_management.TestChannelManagement.test_invite_not_on_channel)
Tests inviting a user to a channel when the inviter is not on that channel. ... DEBUG: Client proposed 'InviterNotOnChan', server assigned 'anonthe_luckyw' with user 'user_anonthe_luckyw' and host 'localhost'. Full response: ['USER user_anonthe_luckyw 0 * :real_anonthe_luckyw', ':localhost 001 anonthe_luckyw :Welcome to the IRC server anonthe_luckyw!', ':localhost 002 anonthe_luckyw :Your host is localhost, running version 1.0', ':localhost 003 anonthe_luckyw :This server was created today', ':localhost 004 anonthe_luckyw localhost 1.0 o o', ':localhost 375 anonthe_luckyw :- ~ Meowdy, traveler! ~', ":localhost 372 anonthe_luckyw :- Here's your good-luck cat butt:", ':localhost 372 anonthe_luckyw :-     /\\_/\\', ':localhost 372 anonthe_luckyw :-    ( o.o )', ':localhost 372 anonthe_luckyw :-     > ^ <    🍑', ':localhost 372 anonthe_luckyw :-     Cat butt initialized.', ':localhost 376 anonthe_luckyw :End of /MOTD']
DEBUG: Client proposed 'InviteeForNC', server assigned 'anonthe_beerdrinkingk' with user 'user_anonthe_beerdrinkingk' and host 'localhost'. Full response: ['USER user_anonthe_beerdrinkingk 0 * :real_anonthe_beerdrinkingk', ':localhost 001 anonthe_beerdrinkingk :Welcome to the IRC server anonthe_beerdrinkingk!', ':localhost 002 anonthe_beerdrinkingk :Your host is localhost, running version 1.0', ':localhost 003 anonthe_beerdrinkingk :This server was created today', ':localhost 004 anonthe_beerdrinkingk localhost 1.0 o o', ':localhost 375 anonthe_beerdrinkingk :- ~ Meowdy, traveler! ~', ":localhost 372 anonthe_beerdrinkingk :- Here's your good-luck cat butt:", ':localhost 372 anonthe_beerdrinkingk :-     /\\_/\\', ':localhost 372 anonthe_beerdrinkingk :-    ( o.o )', ':localhost 372 anonthe_beerdrinkingk :-     > ^ <    🍑', ':localhost 372 anonthe_beerdrinkingk :-     Cat butt initialized.', ':localhost 376 anonthe_beerdrinkingk :End of /MOTD']
DEBUG: Client proposed 'CMemberForInvite', server assigned 'anonthe_mystich' with user 'user_anonthe_mystich' and host 'localhost'. Full response: ['USER user_anonthe_mystich 0 * :real_anonthe_mystich', ':localhost 001 anonthe_mystich :Welcome to the IRC server anonthe_mystich!', ':localhost 002 anonthe_mystich :Your host is localhost, running version 1.0', ':localhost 003 anonthe_mystich :This server was created today', ':localhost 004 anonthe_mystich localhost 1.0 o o', ':localhost 375 anonthe_mystich :- ~ Meowdy, traveler! ~', ":localhost 372 anonthe_mystich :- Here's your good-luck cat butt:", ':localhost 372 anonthe_mystich :-     /\\_/\\', ':localhost 372 anonthe_mystich :-    ( o.o )', ':localhost 372 anonthe_mystich :-     > ^ <    🍑', ':localhost 372 anonthe_mystich :-     Cat butt initialized.', ':localhost 376 anonthe_mystich :End of /MOTD']
Test: anonthe_luckyw inviting anonthe_beerdrinkingk to #inviteonly (inviter not on channel)...
ERROR
test_invite_success (test_channel_management.TestChannelManagement.test_invite_success)
Tests inviting a user to a channel. ... DEBUG: Client proposed 'ChanOpForTest', server assigned 'anonthe_silentq' with user 'user_anonthe_silentq' and host 'localhost'. Full response: ['USER user_anonthe_silentq 0 * :real_anonthe_silentq', ':localhost 001 anonthe_silentq :Welcome to the IRC server anonthe_silentq!', ':localhost 002 anonthe_silentq :Your host is localhost, running version 1.0', ':localhost 003 anonthe_silentq :This server was created today', ':localhost 004 anonthe_silentq localhost 1.0 o o', ':localhost 375 anonthe_silentq :- ~ Meowdy, traveler! ~', ":localhost 372 anonthe_silentq :- Here's your good-luck cat butt:", ':localhost 372 anonthe_silentq :-     /\\_/\\', ':localhost 372 anonthe_silentq :-    ( o.o )', ':localhost 372 anonthe_silentq :-     > ^ <    🍑', ':localhost 372 anonthe_silentq :-     Cat butt initialized.', ':localhost 376 anonthe_silentq :End of /MOTD']
DEBUG: Client proposed 'RegularInviter', server assigned 'anonthe_luckyc' with user 'user_anonthe_luckyc' and host 'localhost'. Full response: ['USER user_anonthe_luckyc 0 * :real_anonthe_luckyc', ':localhost 001 anonthe_luckyc :Welcome to the IRC server anonthe_luckyc!', ':localhost 002 anonthe_luckyc :Your host is localhost, running version 1.0', ':localhost 003 anonthe_luckyc :This server was created today', ':localhost 004 anonthe_luckyc localhost 1.0 o o', ':localhost 375 anonthe_luckyc :- ~ Meowdy, traveler! ~', ":localhost 372 anonthe_luckyc :- Here's your good-luck cat butt:", ':localhost 372 anonthe_luckyc :-     /\\_/\\', ':localhost 372 anonthe_luckyc :-    ( o.o )', ':localhost 372 anonthe_luckyc :-     > ^ <    🍑', ':localhost 372 anonthe_luckyc :-     Cat butt initialized.', ':localhost 376 anonthe_luckyc :End of /MOTD']
DEBUG: Client proposed 'InvitedOne', server assigned 'anonthe_luckyx' with user 'user_anonthe_luckyx' and host 'localhost'. Full response: ['USER user_anonthe_luckyx 0 * :real_anonthe_luckyx', ':localhost 001 anonthe_luckyx :Welcome to the IRC server anonthe_luckyx!', ':localhost 002 anonthe_luckyx :Your host is localhost, running version 1.0', ':localhost 003 anonthe_luckyx :This server was created today', ':localhost 004 anonthe_luckyx localhost 1.0 o o', ':localhost 375 anonthe_luckyx :- ~ Meowdy, traveler! ~', ":localhost 372 anonthe_luckyx :- Here's your good-luck cat butt:", ':localhost 372 anonthe_luckyx :-     /\\_/\\', ':localhost 372 anonthe_luckyx :-    ( o.o )', ':localhost 372 anonthe_luckyx :-     > ^ <    🍑', ':localhost 372 anonthe_luckyx :-     Cat butt initialized.', ':localhost 376 anonthe_luckyx :End of /MOTD']
Test: anonthe_luckyc inviting anonthe_luckyx to #optestchan...
ERROR
test_join_existing_channel_multiple_users (test_channel_management.TestChannelManagement.test_join_existing_channel_multiple_users)
Tests multiple users joining the same channel, and messages propagated correctly. ... Error during recv: [Errno 104] Connection reset by peer
ERROR
test_join_new_channel_success (test_channel_management.TestChannelManagement.test_join_new_channel_success)
Tests if a user can successfully join a new channel. ... ERROR
test_part_channel_success (test_channel_management.TestChannelManagement.test_part_channel_success)
Tests if a user can successfully leave a channel. ... ERROR
test_part_nonexistent_channel (test_channel_management.TestChannelManagement.test_part_nonexistent_channel)
Tests parting a channel that does not exist. ... ERROR
test_part_not_on_channel (test_channel_management.TestChannelManagement.test_part_not_on_channel)
Tests parting a channel that the user is not a member of. ... ERROR
test_topic_get_no_topic (test_channel_management.TestChannelManagement.test_topic_get_no_topic)
Tests getting topic of a channel with no topic set. ... ERROR
test_topic_set_and_get (test_channel_management.TestChannelManagement.test_topic_set_and_get)
Tests setting a topic for a channel and then retrieving it. ... ERROR

======================================================================
ERROR: test_invite_nonexistent_user (test_channel_management.TestChannelManagement.test_invite_nonexistent_user)
Tests inviting a non-existent user.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/test_channel_management.py", line 386, in test_invite_nonexistent_user
    self.assert_irc_reply(response, ERR_NOSUCHNICK, client_nick, nonexistent_nick)
TypeError: IrcTestBase.assert_irc_reply() takes from 3 to 4 positional arguments but 5 were given

======================================================================
ERROR: test_invite_not_on_channel (test_channel_management.TestChannelManagement.test_invite_not_on_channel)
Tests inviting a user to a channel when the inviter is not on that channel.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/test_channel_management.py", line 418, in test_invite_not_on_channel
    self.assert_irc_reply(response, ERR_NOTONCHANNEL, inviter_nick, target_channel)
TypeError: IrcTestBase.assert_irc_reply() takes from 3 to 4 positional arguments but 5 were given

======================================================================
ERROR: test_invite_success (test_channel_management.TestChannelManagement.test_invite_success)
Tests inviting a user to a channel.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/test_channel_management.py", line 356, in test_invite_success
    self.assert_irc_reply(inviter_response, RPL_INVITING, inviter_nick, channel_name, expected_partial_match=invitee_nick)
TypeError: IrcTestBase.assert_irc_reply() got an unexpected keyword argument 'expected_partial_match'

======================================================================
ERROR: test_join_existing_channel_multiple_users (test_channel_management.TestChannelManagement.test_join_existing_channel_multiple_users)
Tests multiple users joining the same channel, and messages propagated correctly.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/test_channel_management.py", line 156, in test_join_existing_channel_multiple_users
    client1_sock.connect((self.SERVER_HOST, self.SERVER_PORT))
ConnectionRefusedError: [Errno 111] Connection refused

======================================================================
ERROR: test_join_new_channel_success (test_channel_management.TestChannelManagement.test_join_new_channel_success)
Tests if a user can successfully join a new channel.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/irc_tester_utils.py", line 173, in setUp
    self.sock.connect((self.SERVER_HOST, self.SERVER_PORT))
ConnectionRefusedError: [Errno 111] Connection refused

======================================================================
ERROR: test_part_channel_success (test_channel_management.TestChannelManagement.test_part_channel_success)
Tests if a user can successfully leave a channel.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/irc_tester_utils.py", line 173, in setUp
    self.sock.connect((self.SERVER_HOST, self.SERVER_PORT))
ConnectionRefusedError: [Errno 111] Connection refused

======================================================================
ERROR: test_part_nonexistent_channel (test_channel_management.TestChannelManagement.test_part_nonexistent_channel)
Tests parting a channel that does not exist.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/irc_tester_utils.py", line 173, in setUp
    self.sock.connect((self.SERVER_HOST, self.SERVER_PORT))
ConnectionRefusedError: [Errno 111] Connection refused

======================================================================
ERROR: test_part_not_on_channel (test_channel_management.TestChannelManagement.test_part_not_on_channel)
Tests parting a channel that the user is not a member of.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/irc_tester_utils.py", line 173, in setUp
    self.sock.connect((self.SERVER_HOST, self.SERVER_PORT))
ConnectionRefusedError: [Errno 111] Connection refused

======================================================================
ERROR: test_topic_get_no_topic (test_channel_management.TestChannelManagement.test_topic_get_no_topic)
Tests getting topic of a channel with no topic set.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/irc_tester_utils.py", line 173, in setUp
    self.sock.connect((self.SERVER_HOST, self.SERVER_PORT))
ConnectionRefusedError: [Errno 111] Connection refused

======================================================================
ERROR: test_topic_set_and_get (test_channel_management.TestChannelManagement.test_topic_set_and_get)
Tests setting a topic for a channel and then retrieving it.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jack/github/ft_irc_dev/irc_tester/tests/irc_tester_utils.py", line 173, in setUp
    self.sock.connect((self.SERVER_HOST, self.SERVER_PORT))
ConnectionRefusedError: [Errno 111] Connection refused

----------------------------------------------------------------------
Ran 10 tests in 31.343s

FAILED (errors=10)
